code/modules/mining/machine_processing.dm

```diff
@@ -5,6 +5,36 @@
 /obj/machinery/mineral
 	var/input_dir = NORTH
 	var/output_dir = SOUTH
+	var/turf/input_turf = null
+	processing_flags = START_PROCESSING_ON_INIT | FAST_PROCESS_SPEED
+
+/obj/machinery/mineral/Initialize(mapload)
+	. = ..()
+	register_input_turf()
+
+/obj/machinery/mineral/Destroy()
+	. = ..()
+	unregister_input_turf()
+
+obj/machinery/mineral/proc/register_input_turf()
+	input_turf = get_step(src, input_dir) // get the turf that players need to place the ore/items onto, defaults to NORTH at round start
+	if(input_turf && istype(input_turf)) // make sure there is actually a turf
+		RegisterSignal(input_turf, COMSIG_ATOM_ENTERED, .proc/begin_processing, TRUE) // listens for when an atom ENTERS the input_turf, tells the machine to begin processing
+
+obj/machinery/mineral/proc/unregister_input_turf()
+	if(input_turf)
+		UnregisterSignal(input_turf, COMSIG_ATOM_ENTERED)
+
+/obj/machinery/mineral/begin_processing()
+	. = ..()
+	// unregistering here prevents `begin_processing` getting called for every item thats dropped onto the `input_turf`
+	// if we did not unregister, that means when someone drops 200 ore stacks, it would call begin_processing 200 times. Not good, and also unnecessary.
+	unregister_input_turf()

```

Comment:
The entire point of being event-based is that you don't need to process for minerals. If this calls begin_processing 200 times because you could potentially drop 200 ores on the turf, then why do you need to process in the first place? You literaly have the 200 events for the items being dropped, use those.

There's no reason for the machine to be processing at all if you're using signals.